src/lib.rs

```diff
@@ -30,28 +27,24 @@ pub fn initialize(filename_prefix: &str) {
         builder.build()
     });
 
-    if level_filter.is_some() {
-        // Ensure the thread local state is always properly initialized.
-        WRITER.with(|rc| {
-            if rc.borrow().is_none() {
-                rc.replace(Some(open_file(filename_prefix)));
-            }
-        });
-    }
-
-    if INITIALIZED.load(atomic::Ordering::Relaxed) || level_filter.is_none() {
+    if level_filter.is_none() {
```

Comment:
This could be further simplified by testing whether the env::var_os result is_none() first, rather than doing map on the Option and testing is_none() here.

Reply:
Yes, that's better, thanks.